File: cdm-sources/cdm/product/collateral/functions/CheckIssuerType.java
package cdm.product.collateral.functions;

import cdm.base.staticdata.asset.common.CollateralIssuerType;
import cdm.product.collateral.EligibilityQuery;
import com.google.inject.ImplementedBy;
import com.rosetta.model.lib.functions.RosettaFunction;
import com.rosetta.model.lib.mapper.MapperS;

import static com.rosetta.model.lib.expression.ExpressionOperators.*;

@ImplementedBy(CheckIssuerType.CheckIssuerTypeDefault.class)
public abstract class CheckIssuerType implements RosettaFunction {

	/**
	* @param issuerType 
	* @param query 
	* @return isEqual 
	*/
	public Boolean evaluate(CollateralIssuerType issuerType, EligibilityQuery query) {
		Boolean isEqual = doEvaluate(issuerType, query);
		
		return isEqual;
	}

	protected abstract Boolean doEvaluate(CollateralIssuerType issuerType, EligibilityQuery query);

	public static class CheckIssuerTypeDefault extends CheckIssuerType {
		@Override
		protected Boolean doEvaluate(CollateralIssuerType issuerType, EligibilityQuery query) {
			Boolean isEqual = null;
			return assignOutput(isEqual, issuerType, query);
		}
		
		protected Boolean assignOutput(Boolean isEqual, CollateralIssuerType issuerType, EligibilityQuery query) {
			isEqual = notExists(MapperS.of(issuerType)).or(contains(MapperS.of(issuerType), MapperS.of(query).<CollateralIssuerType>map("getIssuerType", eligibilityQuery -> eligibilityQuery.getIssuerType()))).get();
			
			return isEqual;
		}
	}
}

File: cdm-sources/src/cdm/product/collateral/functions/CheckIssuerType.py
# pylint: disable=line-too-long, invalid-name, missing-function-docstring, missing-module-docstring, superfluous-parens
# pylint: disable=wrong-import-position, unused-import, unused-wildcard-import, wildcard-import, wrong-import-order, missing-class-docstring
from __future__ import annotations
import sys
import datetime
import inspect
from decimal import Decimal
from rosetta.runtime.utils import *
from rosetta.runtime.func_proxy import replaceable, create_module_attr_guardian
from cdm.product.collateral.EligibilityQuery import EligibilityQuery
from cdm.base.staticdata.asset.common.CollateralIssuerType import CollateralIssuerType

__all__ = ['CheckIssuerType']


@replaceable
def CheckIssuerType(issuerType: CollateralIssuerType | None, query: EligibilityQuery) -> bool:
    """
    
    Parameters 
    ----------
    issuerType : CollateralIssuerType
    
    query : EligibilityQuery
    
    Returns
    -------
    isEqual : boolean
    
    """
    self = inspect.currentframe()
    
    
    isEqual =  ((not rosetta_attr_exists(rosetta_resolve_attr(self, "issuerType"))) or contains(rosetta_resolve_attr(self, "issuerType"), rosetta_resolve_attr(rosetta_resolve_attr(self, "query"), "issuerType")))
    
    
    return isEqual

sys.modules[__name__].__class__ = create_module_attr_guardian(sys.modules[__name__].__class__)
